src/hooks/usePatchStatusSelect.ts

```diff
@@ -113,72 +127,90 @@ export const usePatchStatusSelect = (
       patchStatusFilterTerm !== prevPatchStatusFilterTerm ||
       baseStatusFilterTerm !== prevBaseStatusFilterTerm;
     if (filterTermOrPatchTasksChanged) {
-      if (window.Worker && webWorker) {
-        webWorker.postMessage({
-          patchBuildVariants,
-          patchStatusFilterTerm,
-          baseStatusFilterTerm,
-          selectedTasks,
-        });
-      } else {
-        // fallback in case web workers are not available.
-        // This code reflects logic in public/web_worker/patchBuildVariantReduce.js
-        // Iterate through PatchBuildVariants and determine if a task should be
-        // selected or not based on if the task status correlates with the 2 filters.
-        // if 1 of the 2 filters is empty, ignore the empty filter
-        const baseStatuses = new Set(baseStatusFilterTerm);
-        const statuses = new Set(patchStatusFilterTerm);
-        const nextState =
-          patchBuildVariants?.reduce(
+      // Iterate through PatchBuildVariants and determine if a task should be
+      // selected or not based on if the task status correlates with the 2 filters.
+      // if 1 of the 2 filters is empty, ignore the empty filter
+      let baseStatuses = new Set(baseStatusFilterTerm[versionId]);
+      let statuses = new Set(patchStatusFilterTerm[versionId]);
+
+      let nextState =
+        patchBuildVariants?.reduce(
+          (accumA, patchBuildVariant) =>
+            patchBuildVariant.tasks?.reduce(
+              (accumB, task) => ({
+                ...accumB,
+                [task.id]:
+                  (!!baseStatusFilterTerm[versionId]?.length ||
+                    !!patchStatusFilterTerm[versionId]?.length) &&
+                  (patchStatusFilterTerm[versionId]?.length
+                    ? statuses.has(task.status)
+                    : true) &&
+                  (baseStatusFilterTerm[versionId]?.length
+                    ? baseStatuses.has(task.baseStatus)
+                    : true),
+              }),
+              accumA
+            ),
+          { ...selectedTasks[versionId] }
+        ) ?? {};
```

Comment:
Like wise over here. This appears to be the same code that is contained in the one below we can probably refactor this into something thats easier to read and then extract it into its own function.

Reply:
added to https://jira.mongodb.org/browse/EVG-15414